raiden/network/transport/matrix/transport.py

```diff
@@ -1363,9 +1455,81 @@ def partner_joined(fetched_members: Optional[List[User]]) -> bool:
+        signalling. Messages are exchanged via the partners' private matrix room.
+        Args:
+            rtc_session_description: sdp message for the partner
+            partner_address: Address of the partner
+        """
+        assert self._raiden_service is not None, "_raiden_service not set"
+        if rtc_session_description is None:
+            return
+
+        sdp_type = rtc_session_description.type
+
+        message = {"type": rtc_session_description.type, "sdp": rtc_session_description.sdp}
+        log.debug(
+            f"Send {sdp_type} to partner",
+            node=to_checksum_address(self._raiden_service.address),
+            partner_address=to_checksum_address(partner_address),
+            sdp_description=message,
+        )
+        room = self._get_room_for_address(partner_address, require_online_peer=True)
+        if room is None:
+            return
+
+        if sdp_type == SDPTypes.OFFER.value:
+            self._client.api.invite(room.room_id, message)
+        elif sdp_type == SDPTypes.ANSWER.value:
+            self._client.api.answer(room.room_id, message)
+
+    def create_web_rtc_channel(self, partner_address: Address) -> None:
+        assert self._raiden_service is not None, "_raiden_service not set"
+
+        rtc_partner = self._web_rtc_manager.get_rtc_partner(partner_address)
+        log.debug(
+            "Waiting for initiating web rtc",
+            node=to_checksum_address(self._raiden_service.address),
+            partner_address=to_checksum_address(partner_address),
+        )
+        while (
+            self._get_room_for_address(partner_address, require_online_peer=True) is None
+            or not self._started
+        ):
+            if self._stop_event.is_set():
+                return
+            rtc_partner.partner_ready_event.wait(timeout=5)
+            rtc_partner.partner_ready_event.clear()
+
+        # we can only ask here for capabilities since we have to wait until the user comes online
+        capabilities = self._address_mgr.get_address_capabilities(partner_address)
+        web_rtc_key = Capabilities.WEBRTC.value
+        if web_rtc_key in capabilities and capabilities[web_rtc_key]:
+            log.debug(
+                "Initiating web rtc",
+                node=to_checksum_address(self._raiden_service.address),
+                partner_address=to_checksum_address(partner_address),
+            )
+            self._web_rtc_manager.async_create_channel(partner_address)
+        else:
+            # if no web rtc capabilities by partner remove him from the rtc partners
+            log.debug(
+                "Partner has no web rtc capabilities",
+                node=self._raiden_service.address,
+                partner_address=partner_address,
+                partner_capabilities=capabilities,
+            )
+            del self._web_rtc_manager.address_to_rtc_partners[partner_address]
```

Comment:
Is it certain that the key for the peer will always exist?
Otherwise `.pop(partner_address, None)` should be used to avoid a `KeyError`.

Reply:
line 1502 will create a rtc partner if there is none. But I guess it doesnt hurt to change it here